packages/contracts/README.md

```diff
@@ -70,6 +70,79 @@ Then, copy the deployment address and paste it in to replace `DEPLOYED_CONTRACT_
 npx hardhat verify --network goerli DEPLOYED_CONTRACT_ADDRESS "Hello, Hardhat!"
 ```
 
+## Testing
+
+The `@aragon/osx-versions` package allows you to work with previous versions of the contracts. The solidity source files of these versions are available in the `packages/contracts-versions/versions` directory after the package was built. For a details please refer to the [packages/contracts-versions/README.md](https://github.com/aragon/osx/blob/develop/packages/contracts-versions/README.md).
+
+If you want to import and test a contract from a previous version, you must first import the contract into `contracts/src/tests/osx-versions/Migration.sol`. This step is necessary for the contract to be compiled.
+
+```solidity
+// contracts/src/tests/osx-versions/Migration.sol
+
+import '../../contracts-versions/{version-name}/{path_to_contract}.sol';
+
+// Replace {version-name} with the version number of the contract, and {path_to_contract} with the actual path to the contract.
+```
+
+After successful contract compilation, TypeChain typings will be automatically generated and placed in the `typechain/osx-versions/{version-name}` directory. This will allow you to interact with the contract in a type-safe manner in your tests.
+
+```ts
+// Generic example of usage in a test
+
+...
+
+import {ContractName} from '../../../typechain/osx-versions/{version-name}/{path to ContractName}';
```

Comment:
Less confusing if it's called `MyContract` or something non ambiguous. But no strong feelings in this example

```suggestion
import {MyContract} from '../../../typechain/osx-versions/{version-name}/{path to MyContract}';
```

Reply:
amended